sky/engine/core/dart/window.dart

```diff
@@ -18,41 +24,102 @@ class WindowPadding {
   final double left;
 }
 
+/// An identifier used to select a user's language and formatting preferences,
+/// consisting of a language and a country. This is a subset of locale
+/// identifiers as defined by BCP 47.
 class Locale {
-  Locale(this.languageCode, this.countryCode);
+  const Locale(this.languageCode, this.countryCode);
 
+  /// The primary language subtag for the locale.
   final String languageCode;
+
+  /// The region subtag for the locale.
   final String countryCode;
 
+  bool operator ==(dynamic other) {
```

Comment:
It's just silly that we need to write out this boilerplate every time for equals.  w/o mirrors we can't easily inspect what members this class has, I presume.  Add a field and forget to add it to equals or hashcode and you're a sad panda.